fuel-txpool/src/interface.rs

```diff
@@ -0,0 +1,193 @@
+use crate::{types::*, Config, TxPool as TxPoolImpl};
+use fuel_core_interfaces::block_importer::ImportBlockBroadcast;
+use fuel_core_interfaces::model::{ArcTx, TxInfo};
+use fuel_core_interfaces::txpool::{Error, TxPoolDb, TxPoolMpsc, TxStatus, TxStatusBroadcast};
+use std::cmp::Reverse;
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::{broadcast, mpsc, RwLock};
+
+pub struct Interface {
+    txpool: RwLock<TxPoolImpl>,
+    broadcast: broadcast::Sender<TxStatusBroadcast>,
+    db: Box<dyn TxPoolDb>,
+}
+
+impl Interface {
```

Comment:
This isn't really an interface? Seems like it's all impl with no traits.

Reply:
It acts as an interface between service and txpool, but maybe proxy is a better word

Reply:
yeah, the name is too generic, I'd prefer having some prefix/sufix with it so that it's clear from the name to what exactly it is an interface/proxy

Reply:
It is internally used and not exposed.